hermes/bin/src/ipfs/mod.rs

```diff
@@ -26,48 +26,68 @@ use crate::{
     },
 };
 
-/// Hermes IPFS Internal State
+/// Hermes IPFS Internal Node
 ///
-/// This is a wrapper around `HermesIpfsState` which provides a singleton instance of the
+/// This is a wrapper around `HermesIpfsNode` which provides a singleton instance of the
 /// IPFS state.
 ///
-/// This is done to ensure the IPFS state is initialized only once when the
-/// `HermesIpfsState` is first used. This is done to avoid any issues that may arise if
-/// the IPFS state is initialized multiple times.
+/// This is done to ensure the IPFS Node is initialized only once when the
+/// `HermesIpfsNode` is first used. This is done to avoid any issues that may arise if
+/// the IPFS Node is initialized multiple times.
 ///
-/// The IPFS state is initialized in a separate thread and the sender channel is stored in
-/// the `HermesIpfsState`.
-static HERMES_IPFS_STATE: Lazy<HermesIpfsState> = Lazy::new(|| {
-    let sender = if let Ok(runtime) = Builder::new_current_thread().enable_all().build() {
-        let (sender, receiver) = mpsc::channel(1);
-        let _handle = std::thread::spawn(move || {
-            runtime.block_on(async move {
-                let h = tokio::spawn(ipfs_task(receiver));
-                drop(tokio::join!(h));
-            });
-            std::process::exit(0);
-        });
-        Some(sender)
-    } else {
-        // Failed to start the IPFS task
-        tracing::error!("Failed to start the IPFS task");
-        None
-    };
-    HermesIpfsState::new(sender)
+/// The IPFS Node is initialized in a separate thread and the sender channel is stored in
+/// the `HermesIpfsNode`.
+static HERMES_IPFS: Lazy<HermesIpfsNode> = Lazy::new(|| {
+    let hermes_home_dir = crate::cli::Cli::hermes_home().unwrap_or_else(|err| {
```

Comment:
I think we should have a public free function which will bootstraps hermes ips with the providing arguments in it.
Something similar what we have for event queue `init` function.
To achieve that it would be needed to change this static variable to
```Rust
static HERMES_IPFS: OnceCell<HermesIpfsNode> = OnceCell::new();
```
and the init function could be like this
```Rust
pub(crate) fn bootstrap(dir: &Path) -> anyhow::Result<()> {
    let ipfs_node = HermesIpfsNode::bootstrap(
            IpfsBuilder::new()
            .with_default()
            .set_default_listener()
            .disable_tls()
            .set_disk_storage(ipfs_data_path.clone())
    )?;
    HERMES_IPFS.set(ipfs_node)?;
    Ok(())
}
```
So we will not have a dependency on the `cli` here which is not necessary and it will be possible easier to test things by providing `TempDir` (which will be cleanups automatically).

Also I think it could be a good place of actually instantiating ipfs node for hermes inside `HermesReactor` struct where we already initialising event queue